### PR TITLE
Query Many to Many

### PR DESCRIPTION
**Pull request by cvalencia1991**
# In this milestone, I did the following Activities 

- Understand the different types of relationships between tables.
- Prepare complex queries that answer analytical questions.
- Use primary key & foreign key mechanisms for joining tables.

### CODE REVIEW

**Comment by cvalencia1991** [comment]
Create a table named vets with the following columns:
- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- age: integer
- date_of_graduation: date

image

here is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.

image

There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.

image


Insert the following data for specialties:
- Vet William Tatcher is specialized in Pokemon.
- Vet Stephanie Mendez is specialized in Digimon and Pokemon.
- Vet Jack Harkness is specialized in Digimon.

image


Insert the following data for vets:
- Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
- Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
- Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
- Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

image

Insert the following data for visits:
- Agumon visited William Tatcher on May 24th, 2020.
- Agumon visited Stephanie Mendez on Jul 22th, 2020.
- Gabumon visited Jack Harkness on Feb 2nd, 2021.
- Pikachu visited Maisy Smith on Jan 5th, 2020.
- Pikachu visited Maisy Smith on Mar 8th, 2020.
- Pikachu visited Maisy Smith on May 14th, 2020.
- Devimon visited Stephanie Mendez on May 4th, 2021.
- Charmander visited Jack Harkness on Feb 24th, 2021.
- Plantmon visited Maisy Smith on Dec 21st, 2019.
- Plantmon visited William Tatcher on Aug 10th, 2020.
- Plantmon visited Maisy Smith on Apr 7th, 2021.
- Squirtle visited Stephanie Mendez on Sep 29th, 2019.
- Angemon visited Jack Harkness on Oct 3rd, 2020.
- Angemon visited Jack Harkness on Nov 4th, 2020.
- Boarmon visited Maisy Smith on Jan 24th, 2019.
- Boarmon visited Maisy Smith on May 15th, 2019.
- Boarmon visited Maisy Smith on Feb 27th, 2020.
- Boarmon visited Maisy Smith on Aug 3rd, 2020.
- Blossom visited Stephanie Mendez on May 24th, 2020.
- Blossom visited William Tatcher on Jan 11th, 2021.

image

image

# Query's

- Who was the last animal seen by William Tatcher?

image

- How many different animals did Stephanie Mendez see?

image

- List all vets and their specialties, including vets with no specialties.

image

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.

image

- What animal has the most visits to vets?

image

- Who was Maisy Smith's first visit?

image

- Details for most recent visit: animal information, vet information, and date of visit.

image


- How many visits were with a vet that did not specialize in that animal's species?
image

- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

image

File: data.sql [hunk 1]
```diff
@@ -115,6 +115,57 @@ UPDATE animals
...
+date_of_birth,escape_attempts,
+neutered,weight_kg) VALUES (10,
+'Devimon', '2017-05-12',5,true,11);
+
+--Add specializations Values
+
+INSERT INTO specializations(id,vet_id,spcs_id)
+VALUES (1,1,1),
+        (2,2,1),
+        (3,4,2);
```
Comment (AnselemOdims) [inline data.sql]: - [ ] Good job adding the queries but please kindly include a query to insert into the specializations table where vet name is `Stephanie Mendez` and the specie name is `Digimon`

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @cvalencia1991, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (cvalencia1991) [reply data.sql]: hi, but the question is How many different animals did Stephanie Mendez see not the type of the animals, could you please explain to me, i appreciate it

Comment (cvalencia1991) [reply data.sql]: ohhh i see, thank you im going to fix it

**Review by AnselemOdims — APPROVED** [review]
### Hi  @cvalencia1991 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cvalencia1991** [comment]
thank you very much 👍

**Pull request merged**